qiime2/core/missing.py

```diff
@@ -0,0 +1,104 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2022, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import pandas as pd
+import numpy as np
+
+from .enan import make_nan_with_payload as _make_nan_with_payload
+from .enan import get_payload_from_nan as _get_payload_from_nan
+
+
+def _encode_terms(namespace):
+    enum = _MISSING_ENUMS[namespace]
+    namespace = _NAMESPACE_LOOKUP.index(namespace)
+
+    def encode(x):
+        if type(x) is not str:
+            return x
+        try:
+            code = enum.index(x)
+        except ValueError:
+            return x
+        return _make_nan_with_payload(code, namespace=namespace)
+
+    return encode
+
+
+def _insdc_missing(series):
+    return series.apply(_encode_terms('INSDC:missing'))
+
+
+def _q2_omitted(series):
+    return series
+
+
+def _q2_error(series):
+    if series.isna().any():
+        raise ValueError("Missing values are not allowed in series/column"
+                         " (name=%r) when using scheme 'q2:error'"
+                         % series.name)
+    return series
+
+
+BUILTIN_MISSING = {
+    'INSDC:missing': _insdc_missing,
+    'q2:omitted': _q2_omitted,
+    'q2:error': _q2_error
```

Comment:
I'm thinking a bit about these labels - the `INSDC:missing` works really well in my mind, but the other two feel off. I wonder if the `q2` prefix is a bit too overloaded to be useful in this context? Only because users are reaching for these in a `#q2:missing` directive.

```
INSDC:missing
qiime2:missing
qiime2:all_values_required or qiime2:no_missing_allowed
```

Not too sure about those names, but I'm shooting for a) indicating the INSDC missing values are kinda like the qiime2 ones, just a different vocab, while the last scheme is kinda like our "`error_on_overlap`" choice in feature-table merge. The `q2` vs `qiime2` still doesn't feel great to me. I'll think some more on this.

Reply:
I'm good with `qiime2:missing`, and what about `qiime2:no-missing`? Or `default:blank` and `default:no-missing`?

Reply:
We could even skip the namespace, `blank`/`empty` and `no-missing`?

Reply:
I like `blank` and `no-missing`!

Reply:
Perfect, I'll go with that

Reply:
(and I'm good either way on the namespace, with or without)